backend/routes/admin.py

```
@@ -0,0 +1,656 @@
+            }
+
+            return {"percentiles": percentiles, "sample_size": len(response_times)}
+
+    except Exception as e:
+        logger.error(f"Error fetching response time percentiles: {str(e)}", exc_info=True)
+        raise HTTPException(status_code=500, detail="Error fetching response time percentiles")
+
+
+# Knowledge base endpoints
+@router.get("/knowledge/files")
+async def get_knowledge_files(session: Dict[str, Any] = Depends(require_admin_auth)):
```

Ruff (0.12.2)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)